sktime/contrib/data_io.py

```diff
@@ -0,0 +1,185 @@
+                    meta_data["is_equal_length"] = False
+                elif tokens[1] != "true":
+                    raise IOError(
+                        f"invalid unequal tag value {tokens[1]} in file "
+                        f"{full_file_path_and_name}"
+                    )
+            elif line.startswith("@classlabel"):
+                tokens = line.split(" ")
+                token_len = len(tokens)
+                if tokens[1] == "false":
+                    meta_data["class_labels"] = False
+                elif tokens[1] != "true":
+                    raise IOError(
+                        "invalid classLabel value in file " f"{full_file_path_and_name}"
+                    )
+                if token_len == 2 and meta_data["class_labels"]:
+                    raise IOError(
+                        f"if the classlabel tag is true then class values must be "
+                        f"supplied in file{full_file_path_and_name} but read {tokens}"
+                    )
+            elif line.startswith("@data"):
+                return meta_data
+    raise IOError(
+        f"End of file reached for {full_file_path_and_name} but no indicated start of "
+        f"data with the tag @data"
+    )
+
+
+def load_from_tsfile(
+    full_file_path_and_name,
+    replace_missing_vals_with="NaN",
+    return_y=True,
+):
+    """Load time series data into X and (optionally) y.
+
+    Data from a .ts file into a an 2D (univariate) or 3D (multivariate) if equal
+    length or Pandas DataFrame if unequal length. If present, y is loaded into a 1D
+    array.
+
+    Parameters
+    ----------
+    full_file_path_and_name: str
+        The full pathname of the .ts file to read.
+    replace_missing_vals_with: str, default NaN
+       The value that missing values in the text file should be replaced
+       with prior to parsing.
+    return_y: boolean default True
+       whether to return the y variable, if it is present.
+
+    Returns
+    -------
+    X: DataFrame or ndarray
+    y (optional): ndarray.
+    """
+    # Initialize flags and variables used when parsing the file
+    is_first_case = True
+    instance_list = []
+    class_val_list = []
+    line_num = 0
+    num_dimensions = 0
+    num_cases = 0
+    # equal_length = True
+    with open(full_file_path_and_name, "r", encoding="utf-8") as file:
+        _meta_data = _read_header(file, full_file_path_and_name)
+        for line in file:  # Will this work?
+            num_cases = num_cases + 1
+            line = line.replace("?", replace_missing_vals_with)
+            dimensions = line.split(":")
+            # If first instance then note the number of dimensions.
+            # This must be the same for all cases.
+            if is_first_case:
+                num_dimensions = len(dimensions)
+                if _meta_data["has_class_labels"]:
+                    num_dimensions -= 1
+                for _dim in range(0, num_dimensions):
```

Comment:
Remove for loop here just do:

instance_list = [ [] for _ in range(num_dimensions) ]

Reply:
does it make any difference? this is in contrib, its the wild west :) Still, have changed it